src/network.c

```diff
@@ -1426,12 +1433,7 @@ static int parse_packet (sockent_t *se, /* {{{ */
 			status = parse_part_encr_aes256 (se,
 					&buffer, &buffer_size, flags);
 			if (status != 0)
-			{
-				ERROR ("network plugin: Decrypting AES256 "
-						"part failed "
-						"with status %i.", status);
```

Comment:
My point is that errors happen and it'll be annoying if that means that we're missing error messages. Here, this is easy to avoid, hence my suggestion. I agree that the (-1) message is not overly useful but it will still tell us where something is going wrong. To avoid some duplication, I usually use very specific messages in the helper function and a more generic message in the caller to put the error into context. Here, this could be, for example:

    ERROR: network plugin: Package too short (%d bytes); discarding.
    ERROR: network plugin: Decrypting AES256 part failed (status=%d).

That way, you don't need to repeat "Decryption failed" in all error messages. You'll still get the same amount of information plus the safe-guard against a missing log message. If status would always be -1, we can skip it, or we can make it more useful by actually using different error codes.

What do you think?